### DISCUSSION TITLE
ssp responsible-roles vs responsible-parties

### DISCUSSION THREAD

**Discussion by paulJRCurtis** [usnistgov/OSCAL#1582] [General]
In the ssp `[control-implementation][implemented-requirements]` there is a `responsible-roles` object with the same properties as the `responsible-parties` object in metadata. There are similar examples across other models - quite a few in the assessment layer.

I'm writing reusable components and it would be great to understand if there is there a reason for the different names?

**Comment by GaryGapinski** [comment]
(Edited to correct cardinality error on `responsible-party/party-uuid` — is  {1,∞}.)

I hadn't previously noticed this. They are almost identical except for the element name and child element order. I am uncertain why both exist. Each has a disjoint set of venues "_relative to the containing object_". Both declare **one** role and allow **optional** association to `party` without limit. Since `role` ⇄ `party` association can also occur using `metadata/responsible-party` this allows for semantic uncertainty among such associations (`metadata/responsible-party`, `elsewhere/responsible-role`, `elsewhere/responsible-party`). Both are `role` ⇄ `party` associations.

[`responsible-role`](https://github.com/usnistgov/OSCAL/blob/d3a2b990e24210c253642451e30ea6db99bd045b/xml/schema/oscal_complete_schema.xsd#L1435-L1483) is described as "_A reference to one or more roles with responsibility for performing a function relative to the containing object._". That definition is incorrect, as there can only be one role reference (`@role-id` is not multi-valued).
Has a single required attribute `role-id`.
Allows elements 
1. `prop` {0,∞}
2. `link` {0,∞}
3. `party-uuid` {0,∞}
4. `remarks` {0,1}

(in that order — all are optional).
Occurs in

- `oscal-component-definition-defined-component-ASSEMBLY` {0,∞}
- `oscal-component-definition-implemented-requirement-ASSEMBLY` {0,∞}
- `oscal-component-definition-statement-ASSEMBLY` {0,∞}
- `oscal-implementation-common-system-component-ASSEMBLY` {0,∞}
- `oscal-ssp-implemented-requirement-ASSEMBLY` {0,∞}
- `oscal-ssp-statement-ASSEMBLY` {0,∞}
- `oscal-ssp-by-component-ASSEMBLY` {0,∞}
- `oscal-assessment-common-activity-ASSEMBLY` {0,∞}
- `oscal-assessment-common-task-ASSEMBLY` {0,∞}


[`responsible-party`](https://github.com/usnistgov/OSCAL/blob/d3a2b990e24210c253642451e30ea6db99bd045b/xml/schema/oscal_complete_schema.xsd#L1386-L1434) is described as "_A reference to a set of organizations or persons that have responsibility for performing a referenced role in the context of the containing object._" 

 Has a single required attribute `role-id`. Allows elements 
1. `party-uuid` {1,∞}
2. `prop` {0,∞}
3. `link` {0,∞}
4. `remarks` {0,1}

(in that order).
Occurs in

- `oscal-metadata-metadata-ASSEMBLY` {0,∞}
- `oscal-implementation-common-inventory-item-ASSEMBLY` {0,∞}
- `oscal-ssp-system-characteristics-ASSEMBLY` {0,∞}
- `oscal-assessment-common-assessment-assets-ASSEMBLY` {0,∞}
- `oscal-assessment-common-related-task-ASSEMBLY` {0,∞}
- `oscal-ar-result-ASSEMBLY` {0,∞}

**Reply by CyberDaedalus00** [reply]
If you check cardinality of the party-uuids field in responsible-party, you'll see that it requires at least one party, where the cardinality of the party-uuids field in responsible-role is that the party-uuids is optional .

**Reply by GaryGapinski** [reply]
@CyberDaedalus00 is correct. I had the `party-uuid` cardinality wrong in `responsible-party`. I will edit my response.
